File: betting-backend-nest/src/core/response/login.response.ts
import { Field, ObjectType } from '@nestjs/graphql';

@ObjectType()
export class LoginResponse {
  @Field()
  accessToken: string;

  @Field()
  refreshToken?: string;

  @Field({defaultValue: 'bearer', nullable: true})
  tokenType: string;

  @Field(returns => Number)
  expiresIn: number; // in seconds
}

File: BettingMobileFlutterA/lib/data/model/login_response.dart
class LoginResponse {
  String accessToken;
  String refreshToken;
  String tokenType;
  double expiresIn;
  LoginResponse(
      {this.accessToken, this.refreshToken, this.tokenType, this.expiresIn});

  LoginResponse.fromJson(Map<String, dynamic> json) {
    accessToken = json['accessToken'];
    refreshToken = json['refreshToken'];
    tokenType = json['tokenType'];
    expiresIn = json['expiresIn'].toDouble();
  }

  Map toJson() {
    Map data = {};
    data['accessToken'] = accessToken;
    data['refreshToken'] = refreshToken;
    data['tokenType'] = tokenType;
    data['expiresIn'] = expiresIn;
    return data;
  }
}
